utils/config_file.hh

```diff
@@ -66,9 +66,15 @@ public:
     typedef std::vector<sstring> string_list;
 
     enum class value_status {
-        Used,
-        Unused,
-        Invalid,
+        Used, ///< supported by Scylla. only the "Used" options are added to the
+              ///< command line options, and can be specified with command line.
+        Unused, ///< no-more supported.  we just silently ignore these options
+                ///< at seeing them. typically used for options supported by the
```

Comment:
at seeing them **in the YAML configuration file**, right? As you said above, they can't be used as command-line parameters, right?

Reply:
Intention was:

Used - a valid option that has impact on stuff
Unused - an option inherited or deprecated, or not implemented (yet). In the initial translation from cassandra we had loads of those - options that were supposed to exist and work, but did not yet. 
Invalid - options inherited or deprecated that we want an explicit warning (or error) if given. Looking at config.hh it seems we have a little to many of them imho...

Reply:
thank you @elcallio . i incorporated your reply into the change with a little bit editing.